### ISSUE TITLE
problem with scroll after <TAB><TAB> if the filename is too long

### ISSUE THREAD

**Issue by simonbcn** [fish-shell/fish-shell#2557]
Ubuntu 14.04 64bits
fish v2.2.0 (installed from `ppa:fish-shell/release-2`)
terminator 0.97

I hit `TAB` twice to cycle through the possibilities but it does a wrong scroll and it deletes my command:

[![asciicast](https://asciinema.org/a/b9fcyzd72agnuytpl70zfxkqr.png)](https://asciinema.org/a/b9fcyzd72agnuytpl70zfxkqr)

**Comment by pickfire** [comment]
Try using the latest fish, the fish version that you are using have bad completion:

 - The highlight line stays when you had switched to a new one.

**Comment by simonbcn** [comment]
I've compiled and installed the git version and same problem.

**Comment by pickfire** [comment]
I cannot reproduce that.

**Comment by simonbcn** [comment]
I've tested with XTerm, LXTerminal and gnome-terminal. In all of them it fails.

**Comment by pickfire** [comment]
I use st, but I don't find any problem.

**Comment by simonbcn** [comment]
Tested with st and same problem.

**Comment by pickfire** [comment]
I can't reproduce it, I am using arch on arm architecture.

**Comment by simonbcn** [comment]
PC Intel 64 bits.

**Comment by zanchey** [comment]
That is strange. What is your locale set to? The output of `locale` may be useful.

**Comment by simonbcn** [comment]
Same problem with Arch Linux 64 bits. 
```
$ locale
LANG=es_ES.utf8
LC_CTYPE="es_ES.utf8"
LC_NUMERIC="es_ES.utf8"
LC_TIME="es_ES.utf8"
LC_COLLATE="es_ES.utf8"
LC_MONETARY="es_ES.utf8"
LC_MESSAGES="es_ES.utf8"
LC_PAPER="es_ES.utf8"
LC_NAME="es_ES.utf8"
LC_ADDRESS="es_ES.utf8"
LC_TELEPHONE="es_ES.utf8"
LC_MEASUREMENT="es_ES.utf8"
LC_IDENTIFICATION="es_ES.utf8"
LC_ALL=
```

**Comment by faho** [comment]
@simonbcn: Which completion are you having this issue with? All of them? Just a file completion? Because in the original report there's `makemkvcon`, which I don't know and can't find and don't know if we have a completion for, and some unusual file/directory names (lots of spaces and parens).

It's also possible that it's a purely visual issue with file completion where a single file runs longer than a line.

**Comment by faho** [comment]
I get something like this, and as far as I can tell it _is_ related to completion options using more than one line. @simonbcn: Does this improve if you make your terminal wider?

**Comment by simonbcn** [comment]
It fails in ttyX consoles too.
It doesn't fail always, only with certain folders and it isn't related with completion of any program.
For example:
[![asciicast](https://asciinema.org/a/evy6duwnifzumh3x2sraf8wsr.png)](https://asciinema.org/a/evy6duwnifzumh3x2sraf8wsr)

In this example it doesn't fail:
[![asciicast](https://asciinema.org/a/4px26rwaxbrl7dovsqttmn6jn.png)](https://asciinema.org/a/4px26rwaxbrl7dovsqttmn6jn)

**Comment by faho** [comment]
@simonbcn: Look carefully - in your failed example, it goes over the line. That doesn't happen in the passing one. And I'll bet that's what triggers it.

**Comment by simonbcn** [comment]
ok, that is the bug. I change the title of this issue.

**Comment by floam** [comment]
Does this still happen?

**Comment by jusga** [comment]
Steps to reproduce this indepent of terminal size with `fish, version 2.5.0-259-gc0de8afa`:

- Start with an empty directory and create some files:
  ```
  > touch file(seq -w 1 $LINES)
  ```
- Then we need an file with a long name:
  ```
  > touch (string repeat -n $COLUMNS a)
  ```
- Type some command with filename completion and hit `<Tab>` repeatedly.

[![asciicast](https://asciinema.org/a/clr9ti4ot00xd5c7tg5uyzfi8.png)](https://asciinema.org/a/clr9ti4ot00xd5c7tg5uyzfi8)

**Comment by faho** [comment]
I can no longer reproduce this, closing.

**Comment by zanchey** [comment]
It's broken in 3.1.2 but working in master, so I'll mark this as fixed in 3.2.0.

**Comment by krobelus** [comment]
Sounds like ada03d350 (Correct pager size when command line soft-wraps)
